### DISCUSSION TITLE
Logs: Add the possibility to permalink to a certain logline

### DISCUSSION THREAD

**Discussion by svennergr** [grafana/grafana#56093] [Observability]
It would be great to have a functionality to permalink to a certain logline.

**Comment by gabor** [comment]
if we decide to implement this, we should keep in mind that some logs-databases are not able to "load a log line based on an id", so in those cases we might have to go with "load all log lines for a given timestamp", or somehow get smart :)

**Comment by gabor** [comment]
i wonder... should this work like the log-context-thing? i mean, if you open that permalink in grafana, what should appear?
just the single log-line (i don't think this is what is needed, because for this one we can just store the log-line in the query-string 😁 )? 
maybe it should show up the way that the permalinked log-line is highlighed in the middle and a couple log-lines "above" and "below"?

**Reply by svennergr** [reply]
Yea that's a good point. That's why I thought we might base this around the log-context components. With the possibility to first open the log-context in a split view, which could be more easily shared.

**Reply by 09jvilla** [reply]
@gabor - what you described is along the lines of what I was thinking. It would be the exact same experience I have today of sharing a URL which reruns a specific query, but in addition to rerunning that query and showing those results, I show those results with a specific log line highlighted.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
